src/Termonad/Types.hs

```diff
@@ -546,70 +593,63 @@ invariantTMState' tmState =
     runInvariants = fmap catMaybes . sequence
 
     invariantFocusSame :: IO (Maybe TMStateInvariantErr)
-    invariantFocusSame = do
+    invariantFocusSame = execExceptT $ do
```

Comment:
I switched the invatiant-checking code to use ExceptT, because it simplified the logic when I had repeated checks which could fail. However, I no longer have repeated checks (I used to check that the notebook page was a Paned and that its first child was the expected ScrolledWindow, but now I simply check that the notebook page is the expected Pane), so it would be easy to switch back.

Reply:
That seems reasonable.

Once we figure out how to represent the `Tree`(?) of Panes, I imagine we should add another invariant to make sure the Pane Tree in a Tab is represented correctly in `tmState`.  But I imagine this would probably come near the end of the PR, or possibly in a follow-up PR.